addon/hint/show-hint.js

```diff
@@ -268,8 +268,10 @@
     var winH = parentWindow.innerHeight || Math.max(ownerDocument.body.offsetHeight, ownerDocument.documentElement.offsetHeight);
     container.appendChild(hints);
 
-    var box = completion.options.moveOnOverlap ? hints.getBoundingClientRect() : new DOMRect();
     var scrolls = completion.options.paddingForScrollbar ? hints.scrollHeight > hints.clientHeight + 1 : false;
+    if (scrolls) for (var node = hints.firstChild; node; node = node.nextSibling)
+      node.style.paddingRight = cm.display.nativeBarWidth + "px"
```

Comment:
Why did you move this code? This ordering forces another relayout.

Reply:
Before my change the code would check if the hints container is off screen. If so, it would move it to be aligned with the edge of the screen. After that, it checks if the hints container has a scrollbar and if so, it applies padding to each hint element which causes the container to grow and be off screen again. (see attached gif)

I just changed the order so that the padding for the scrollbar is applied before it is checked if the container is off screen.

![2021-03-25_17-53-49](https://user-images.githubusercontent.com/9429273/112512130-61799900-8d93-11eb-8476-60099f0932d4.gif)

Reply:
Yes, but this code has been carefully built to _not_ create unnecessary relayouts, and I don't really want to accept a patch that degrades.

Reply:
I don't get why it should create another relayout, it just moves the container once in any case, or am I missing something?

Reply:
See for example [https://eloquentjavascript.net/14_dom.html#p_LcMNzqCMUS](this link). Your change causes a write to the DOM to happen between two layout reads, which will force an extra relayout.

Reply:
I see... I actually read your book but didn't remember that section.
Didn't think of that! 

I did some measurements and my change would have added around 10ms of rendering time.
I now have a more elegant solution: If there are scrollbars, the amount of padding added to the items is just added to the `overlapX`value, so no re-layout necessary anymore.
My measurements confirm that it adds no additional rendering time this way:

![image](https://user-images.githubusercontent.com/9429273/112617479-5111fe80-8e25-11eb-8dcb-8d657d4b564e.png)
![image](https://user-images.githubusercontent.com/9429273/112617491-54a58580-8e25-11eb-8740-653a129e579f.png)